pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageRouterBase.java

```diff
@@ -19,11 +19,19 @@
 package org.apache.pulsar.client.impl;
 
 import org.apache.pulsar.client.api.MessageRouter;
+import org.apache.pulsar.client.api.ProducerConfiguration;
 
 public abstract class MessageRouterBase implements MessageRouter {
-    final boolean useMurmurHash;
+    protected final Hash hash;
 
-    MessageRouterBase(boolean useMurmurHash) {
-        this.useMurmurHash = useMurmurHash;
+    MessageRouterBase(ProducerConfiguration.HashingScheme hashingScheme) {
+        switch (hashingScheme) {
+            case JavaStringHash:
+                this.hash = new JavaStringHash();
```

Comment:
Maybe we can keep static instances of the Hashing objects since they can be shared around